core/network/src/p2p/gossip/handler.rs

```diff
@@ -0,0 +1,107 @@
+use std::error::Error;
+use std::fmt::Debug;
+use tokio::sync::Mutex;
+use std::sync::Arc;
+use std::time::Duration;
+use async_trait::async_trait;
+use prost::Message;
+use avalanche_types::ids::node::Id;
+use crate::p2p;
+use crate::p2p::gossip::{Gossipable, Set};
+use crate::p2p::sdk::{PullGossipRequest, PullGossipResponse};
+
+pub struct HandlerConfig {
+    pub namespace: String,
+    pub target_response_size: usize,
+}
+
+#[derive(Debug, Clone)]
+pub struct Handler<S: Set + Debug> {
+    pub set: Arc<Mutex<S>>,
+    pub target_response_size: usize,
+
+}
+
+pub fn new_handler<S: Set + Debug>(
+    config: HandlerConfig,
+    set: Arc<Mutex<S>>,
+) -> Handler<S> {
+    Handler {
+        set,
+        target_response_size: config.target_response_size,
+    }
+}
+
+#[async_trait]
+#[allow(unused_variables)]
```

Comment:
You shouldn't really ever have this on a trait. Just use the `_` prefix where necessary.